bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergenieHandler.java

```diff
@@ -0,0 +1,183 @@
+                logger.debug("Couldn't get I/O for the connection to: {}:{}.", host, TCP_PORT, e);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        final EnergenieConfiguration config = getConfigAs(EnergenieConfiguration.class);
+
+        if (config.host != null && config.password != null) {
+            refreshInterval = EnergenieConfiguration.DEFAULT_REFRESH_INTERVAL;
+            host = config.host;
+            logger.debug("Initializing EnergenieHandler for Host '{}'", config.host);
+            energenieSocket = new EnergenieSocket(config.host, config.password);
+
+            updateStatus(ThingStatus.UNKNOWN);
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refreshState, 5, refreshInterval, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device , IP-Address or password not set");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("EnergenieHandler disposed.");
+        final ScheduledFuture<?> refreshJob = this.refreshJob;
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    private @Nullable Boolean refreshState() {
+        final EnergenieSocket socket = this.energenieSocket;
+
+        if (socket != null) {
+            try {
+                stateUpdate(socket.retrieveStatus());
+                if (thing.getStatus() != ThingStatus.ONLINE && ThingHandlerHelper.isHandlerInitialized(thing)) {
+                    updateStatus(ThingStatus.ONLINE);
+                }
+                return Boolean.TRUE;
+            } catch (final UnknownHostException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Can't find host");
+                logger.debug("Can't find host: {}:{}.", host, TCP_PORT, e);
+            } catch (final IOException e) {
+                logger.debug("Couldn't get I/O for the connection to: {}:{}.", host, TCP_PORT, e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Couldn't get I/O for the connection");
+            } catch (final RuntimeException e) {
+                logger.debug("Unexpected error", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
+            }
+        }
+        return null;
+    }
+
+    public void stateUpdate(final byte[] status) {
+        for (int i = 0; i < 4; i++) {
+            final String socket = CHANNEL_SOCKET_PREFIX + (i + 1);
+            final String stringStatus = String.format("0x%02x", status[i]);
+
+            if (stringStatus.equals(statusOn)) {
+                updateState(socket, OnOffType.ON);
+            } else if (stringStatus.equals(statusOff)) {
+                updateState(socket, OnOffType.OFF);
```

Comment:
What if there is another value? Would it make sense to set `UNDEF` then? Or - if other values are not possible:

```
    updateState(socket, OnOffType.from(stringStatus.equals(statusOn));

Reply:
Done